subsys/caf/modules/sensor_sampler.c

```diff
@@ -223,7 +227,11 @@ static void sample_sensor(struct sensor_data *sd, const struct sensor_config *sc
 		LOG_ERR("Sensor sampling error (err %d)", err);
 		update_sensor_state(sc, sd, SENSOR_STATE_ERROR);
 	} else {
-		send_sensor_event(sc->event_descr, curr, ARRAY_SIZE(curr));
+		if (atomic_get(&sd->active_sensor_events_cnt) <= sd->active_sensor_events_limit) {
+			send_sensor_event(sc->event_descr, curr, ARRAY_SIZE(curr), sd);
+		} else {
+			LOG_INF("Did not send event due to too many active events");
```

Comment:
Log warning (since probably it's not expected situation)?
Also consider informing for which sensor the problem occurred.